application/src/main/java/org/togetherjava/tjbot/commands/system/BotCore.java

```diff
@@ -94,54 +99,121 @@ public BotCore(JDA jda, Database database, Config config) {
             .toList();
 
         // User Interactors (e.g. slash commands)
-        nameToInteractor = features.stream()
+        interactors = features.stream()
             .filter(UserInteractor.class::isInstance)
             .map(UserInteractor.class::cast)
-            .collect(Collectors.toMap(UserInteractor::getName, Function.identity()));
+            .filter(validateInteractor())
+            .toList();
+
+        nameToInteractor = interactors.stream().collect(Collectors.toMap(interactor -> {
+            String name = interactor.getName();
+
+            if (interactor instanceof SlashCommand) {
+                return "s-" + name;
+            } else if (interactor instanceof MessageContextCommand) {
+                return "mc-" + name;
+            } else if (interactor instanceof UserContextCommand) {
+                return "uc-" + name;
+            }
```

Comment:
this prefix logic has to be extracted into a single place. Id vote for a `record InteractorId(String prefix, String commandName)` which has a factory for being constructed out of the interactor `static InteractorId ofInteractor(Interactor interactor)` or sth like that.

otherwise we will likely duplicate the prefixes at multiple places and wont easily spot where to do adjustments if we want to add new stuff